frontend/src/communications/userApi.js

```diff
@@ -1,7 +1,13 @@
 import axios from "axios";
-import { fetchMockUsersForListing } from "./mockForUserApi";
+import mockApi from "./mockForUserApi";
 
-export const fetchUsersWithApi = () => {
-  fetchMockUsersForListing();
-  return axios.get("/users");
+export const getUsers = async (setUsers, setError) => {
+  try {
+    await axios.get("/users").then((response) => {
+      setUsers(response.data.users);
```

Comment:
Szebb lenne a hibakezelés, ha nem callbackeket kapnánk. Ha sikerült, akkor sima return response.data.users, ha nem, akkor meg lehet dobni tovább egy saját exceptiont, például: throw new Error("Failed to get users")

Reply:
Ebben nem vagyok biztos, hogy jól sikerült-e megvalósítanom a javaslataid.